src/extension/tab/tab.ts

```diff
@@ -50,5 +50,12 @@ if (__IS_FIREFOX__) {
     }
   }
 
-  injectScript(browser.runtime.getURL("hook.js"));
+  try {
+    injectScript(browser.runtime.getURL("hook.js"));
+  } catch (e) {
+    console.warn(
+      "Error occured while trying to initialize Apollo Client Devtools:",
+      e
+    );
+  }
```

Comment:
This doesn't work, as the error is triggered delayed when the script evaluates, not when it is inserted in DOM.

Also, listening for a `securitypolicyviolation` event allows me to read the details of the violation, but not to prevent it from crashing the whole page. I'm a bit out of ideas.